spiff/__main__.py

```diff
@@ -1,2 +1,127 @@
+import json
+import logging
+import os
+import sys
+from datetime import datetime
+
+import coolname
+import torch
+from pytorch_lightning import Trainer
+from pytorch_lightning import loggers as pl_loggers
+
+import spiff.models as models
+from chem.sim import SCSimilarity
+from data.datamodules import ZincDatamodule
+from data.featurizer import GraphFeaturizerFactory
+from spiff.cfg import ExperimentConfig
+from spiff.cli import override_with_flags, parse_arguments
+from spiff.experiments import SPiFFModule
+from spiff.mining import TripletMiner
+
+logger = logging.getLogger(__name__)
+
+
+def get_run_name(base: str) -> str:
+    if base:
+        name = base
+    else:
+        name = coolname.generate_slug(2)
+    return f"{name}-{datetime.now().strftime('%d.%m.%Y-%H:%M')}"
+
+
+def set_up_experiment_logger(
+    config: ExperimentConfig, name: str, use_wandb: bool
+) -> pl_loggers.Logger:
+    hyperparams = {
+        "learning_rate": config.learning_rate,
+        "batch_size": config.batch_size,
+    }
+    hyperparams.update(config.model_config.__dict__)
+    if use_wandb:
+        os.makedirs(cfg.system_config.results_dir, exist_ok=True)
+        wandb_logger = pl_loggers.WandbLogger(
+            project=config.system_config.wandb_project,
+            entity=config.system_config.wandb_entity,
+            name=name,
+            save_dir=cfg.system_config.results_dir,
+        )
+        wandb_logger.experiment.config.update(hyperparams)
+        return wandb_logger
+    else:
+        results_dir = os.path.join(config.system_config.results_dir, "csv")
+        csv_logger = pl_loggers.CSVLogger(save_dir=results_dir, name=name)
+        csv_logger.log_hyperparams(hyperparams)
+        return csv_logger
+

```

Comment:
Maybe it's possible to disable logging (and checkpointing) completely? for a 'true' debug run.